core/src/main/scripts/importer/importOncokbMutation.py

```diff
@@ -0,0 +1,293 @@
+    logger.addHandler(collapsing_text_handler)
+
+    meta_file_mutation = libImportOncokb.find_meta_file_by_fields(study_dir, {'genetic_alteration_type': 'MUTATION_EXTENDED'})
+    mutation_file_path = os.path.join(study_dir, libImportOncokb.find_data_file_from_meta_file(meta_file_mutation))
+    check_required_columns(libImportOncokb.get_first_line(open_mutations_file(mutation_file_path)).rstrip('\n').split('\t'))
+    check_disallowed_columns(libImportOncokb.get_first_line(open_mutations_file(mutation_file_path)).rstrip('\n').split('\t'))
+
+    global portal_instance
+    if hasattr(args, 'portal_info_dir') and args.portal_info_dir is not None:
+        portal_instance = validateData.load_portal_info(args.portal_info_dir, logger,
+                                           offline=True)
+    else:
+        portal_instance = validateData.load_portal_info(server_url, logger)
+
+    row_number_to_feature = get_features(mutation_file_path)
+    row_number_to_annotation = fetch_oncokb_annotations(row_number_to_feature)
+    write_annotations_to_file(row_number_to_annotation, mutation_file_path)
+
+    logger.info('Import complete')
+
+    return exit_status_handler.get_exit_status()
+
+
+def open_mutations_file(file_name):
+    """Open MAF file and handle exception when not found."""
+    try:
+        file = open(file_name)
+    except FileNotFoundError:
+        raise FilenotFoundError("Could not open MAF file at path '" + file_name + "'")
+    return file
+
+
+def check_disallowed_columns(header_elements):
+    disallowed_columns = []
+    for disallowed_column in disallowed_mutation_columns:
+        if disallowed_column in header_elements:
+            disallowed_columns.append(disallowed_column)
+    if len(disallowed_columns) > 0:
+        raise RuntimeError("One or more disallowed columns for OncoKb import are present in the MAF file. " \
+                           "Disallowed column(s): [" + ", ".join(disallowed_columns) + "]")
+
+
+def get_features(mutation_file_path):
+    """Extract Mutation events from MAF data file."""
+    header_elements = libImportOncokb.get_first_line(open_mutations_file(mutation_file_path)).rstrip().split('\t')
+    header_indexes = {}
+    for required_column in required_mutation_columns + ['Entrez_Gene_Id']:
+        header_indexes[required_column] = header_elements.index(required_column)
+    row_number_to_feature = {}
+    row_counter = 0
+    mutation_file = open_mutations_file(mutation_file_path)
+    print("Reading features from file ...", end = '')
+    for line in mutation_file:
+        row_counter += 1
+        if line == '\n' or line.startswith('#') or line.startswith(header_elements[0]):
+            continue  # skip comment and header line
+        line_elements = line.rstrip().split('\t')
+        feature = {}
+        for column_name, index in header_indexes.items():
+            value = line_elements[index]
+            if value != '':
+                if column_name == 'HGVSp_Short':
```

Comment:
It would be great to put this as a string constant.

Reply:
Done